### ISSUE TITLE
azurerm_windows_web_app: autoheal settings "sub_status" and "win32_status" are not set

### ISSUE THREAD

**Issue by pietersap** [hashicorp/terraform-provider-azurerm#17862]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_windows_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_windows_web_app" "api" {
    name = "acc-${var.project_name}-app-${terraform.workspace}-${var.resource_suffix}"
    location = data.azurerm_resource_group.project_resource_group.location
    resource_group_name = data.azurerm_resource_group.project_resource_group.name
    service_plan_id = data.azurerm_service_plan.asp.id

    https_only = true

    app_settings = local.app_settings

    identity {
        type = "SystemAssigned"
    }

    site_config {
        ftps_state = "Disabled"
        always_on = true
        health_check_path = "/health/live"

        auto_heal_enabled = true
        auto_heal_setting {
            trigger {
                status_code {
                  count = 1
                  status_code_range = 500
                  sub_status = 37
                  interval = "00:01:00"
                }
                status_code {
                  count = 1
                  status_code_range = 500
                  sub_status = 30
                  win32_status = 0
                  interval = "00:10:00"                 
                }
            }
            action {
                action_type = "Recycle"
            }
        }
    }
}
```

### Debug Output/Panic Output

```shell
Apply complete!
```

### Expected Behaviour

The sub_status and win32_status are set correctly. When re-applying the Terraform configuration, the configuration matches the infrastructure. Values should be set correctly: sub_status = 30 or 37, and win32_status = 0.

### Actual Behaviour

The console output tells me that all changes have been applied correctly (see output above). 

But the ```sub_status ``` and  ```win32_status```  values are not set correctly and they appear as "0" (zero) when checking in the Azure Portal.

![image](https://user-images.githubusercontent.com/32063689/182567185-170bca03-a7c9-4e3e-817f-449fcd92f2c8.png)

When re-applying Terraform, the plan proposes to set the values again. Etcetera. (see output below: Terraform tries - again - to set sub_status and win32_status. Note that the plan proposes to _change_ the sub_status from 0 to 30/37, and to _add_ the win32_status, but it does not work and it repeats at the next apply.

```
  # module.hr_api_webapp_aggregator.azurerm_windows_web_app_slot.staging will be updated in-place
  ~ resource "azurerm_windows_web_app_slot" "staging" {
        id                                = "/subscriptions/4fede0c0-65b2-49c2-8832-905bfeaf4d2f/resourceGroups/boot-dev-rg/providers/Microsoft.Web/sites/acc-boot-app-dev-agg/slots/staging"     
        name                              = "staging"
        tags                              = {}
        # (18 unchanged attributes hidden)



      ~ site_config {
            # (23 unchanged attributes hidden)


          ~ auto_heal_setting {

              ~ trigger {
                    # (1 unchanged attribute hidden)

                  ~ status_code {
                      ~ sub_status        = 0 -> 37
                        # (3 unchanged attributes hidden)
                    }
                  ~ status_code {
                      ~ sub_status        = 0 -> 30
                      + win32_status      = "0"
                        # (3 unchanged attributes hidden)
                    }
                }
                # (1 unchanged block hidden)
            }
            # (1 unchanged block hidden)
        }
        # (2 unchanged blocks hidden)
    }
```

### Steps to Reproduce

1: terraform apply
2: check in portal. Sub_status and win32_status are not set
3: terraform apply again
4. etc.

**Comment by xiaxyi** [comment]
Hi @pietersap , the fix is included in the pr https://github.com/hashicorp/terraform-provider-azurerm/pull/21831, you may track it for a progress. Feel free to let me know if you have any questions.

**Comment by pietersap** [comment]
@xiaxyi great, thank you! Do I understand correctly that this will be included in the 3.63 version? I will validate ASAP when the version is available

**Comment by pietersap** [comment]
@xiaxyi @jackofallops  Thanks for the effort and updates. Confirmed that it's fixed in AzureRM 3.63.0
